sources/api/schnauzer/src/helpers.rs

```diff
@@ -1328,6 +1339,216 @@ pub fn etc_hosts_entries(
+/// ["one", "two"] the returned value will be ["one", "two"] which otherwise will be written as
+/// [one, two] with plain handlebars.
+pub fn json_string(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    // To give context to our errors, get the template name, if available.
+    trace!("Starting json_string helper");
+    let template_name = template_name(renderctx);
+    trace!("Template name: {}", &template_name);
+
+    // Check number of parameters, must be exactly one
+    trace!("Number of params: {}", helper.params().len());
+    check_param_count(helper, template_name, 1)?;
+
+    let value = helper
+        .param(0)
+        .map(|v| v.value())
+        .context(error::ParamUnwrapSnafu {})?
+        .to_string();
+
+    // Write the value as a JSON string
+    out.write(&value).context(error::TemplateWriteSnafu {
+        template: template_name.to_owned(),
+    })?;
+
+    Ok(())
+}
+
+/// This helper returns the valid values for the ECS metadata service limits
+///
+/// This helper returns the properly formatted ECS metadata service limits. When either RPS, burst
+/// or both are missing, the default values in the ECS agent are used instead.
+pub fn ecs_metadata_service_limits(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    // To give context to our errors, get the template name, if available.
+    trace!("Starting ecs_metadata_service_limits helper");
+    let template_name = template_name(renderctx);
+    trace!("Template name: {}", &template_name);
+
+    // Check number of parameters, must be exactly two (metadata_service_rps and
+    // metadata_service_burst)
+    trace!("Number of params: {}", helper.params().len());
+    check_param_count(helper, template_name, 2)?;
+
+    let metadata_service_rps = helper
+        .param(0)
+        .map(|v| v.value())
+        .context(error::ParamUnwrapSnafu {})?;
+
+    let metadata_service_burst = helper
+        .param(1)
+        .map(|v| v.value())
+        .context(error::ParamUnwrapSnafu {})?;
+
+    let output = match (metadata_service_rps, metadata_service_burst) {
+        (Value::Number(rps), Value::Number(burst)) => {
+            format!("{},{}", rps, burst)
+        }
+        (Value::Number(rps), Value::Null) => {
+            format!("{},{}", rps, DEFAULT_ECS_METADATA_SERVICE_BURST)
+        }
+        (Value::Null, Value::Number(burst)) => {
+            format!("{},{}", DEFAULT_ECS_METADATA_SERVICE_RPS, burst)
+        }
+        // It is unlikely that this is reached with values of type different than Null, since the
+        // API validates the types
+        (_, _) => format!(
+            "{},{}",
+            DEFAULT_ECS_METADATA_SERVICE_RPS, DEFAULT_ECS_METADATA_SERVICE_BURST
+        ),
```

Comment:
This comment is only true if the template is well-formed and only number-valued settings are passed to the helper. IMO it should be a rendering error instead for unknown types.

But you do need the `(Value::Null, Value::Null)` case.